test/index.spec.js

```diff
@@ -110,5 +111,36 @@ describe('immutableStateInvariantMiddleware', () => {
     expect(() => {
       dispatch({type: 'SOME_ACTION'});
     }).toNotThrow();
-  })
+  });
+
+  it('logs if mutating inside the dispatch', () => {
```

Comment:
Can we wrap these two `it`s to its own describe and use `beforeEach` and `afterEach` to mock and restore the `console.error`? That would handle intermediate errors in a test and still restore the mock for other tests.